### PR TITLE
Astro CLI `build-secret` documentation

### PR DESCRIPTION
**Pull request by jwitz**
This needs thorough technical review, as my instructions here are mostly guesswork based on the Docker documentation.

### CODE REVIEW

File: astro/cli/astro-dev-start.md [hunk 1]
```diff
@@ -27,7 +27,9 @@ astro dev start
...
 | ----------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | ----------------------------------------------------------------------------------- |
+| `--build-secrets` | Run `docker build --secret` to mount a secret value to your Docker image. | `id=<your-secret-id>, src=<path-to-secret> .` See [Docker documentation](https://docs.docker.com/build/building/secrets/#secret-mounts). |
 | `--compose-file`        | The location of a custom Docker Compose file to use for starting Airflow on Docker.                                                                                                                                                                                | Any valid filepath                                                                  |
+
```
Comment (jwitz) [inline astro/cli/astro-dev-start.md:32]:
```suggestion
```

File: astro/cli/private-python-packages.md [hunk 2]
```diff
@@ -164,7 +164,17 @@ Make sure that the name of any privately hosted Python package doesn’t conflic
 
 :::
 
-#### Step 2: Update Dockerfile
+#### Step 2: Define your extra index URL
+
+Create a new file called `indexurl.txt` that contains the extra index URL used to access your private PuPI index. For example:
+
+```sh
```
Comment (jwitz) [inline astro/cli/private-python-packages.md:171]:
```suggestion
```text
```

File: astro/cli/private-python-packages.md [hunk 3]
```diff
@@ -164,7 +164,17 @@ Make sure that the name of any privately hosted Python package doesn’t conflic
 
 :::
```
Comment (jwitz) [inline astro/cli/private-python-packages.md:166]: Can this command also be used to optimize the previous process (Private GitHub Repo installation)? I was having trouble figuring out what format the secret would be in. @sunkickr

File: astro/cli/astro-dev-upgrade-test.md [hunk 4]
```diff
@@ -33,6 +33,7 @@ By default, the command runs all three available tests on your project against t
...
 | ---------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------ |
 | `-a`, `--airflow-version`    | The equivalent of Airflow you want to upgrade to. The default is the latest available version. Note that the Astro CLI will still test against an Astro Runtime image based on the Airflow version you specify.                         | Any valid [Airflow version](https://airflow.apache.org/docs/apache-airflow/stable/release_notes.html). |
+| `--build-secrets` | Run `docker build --secret` to mount a secret value to your Docker image. | `id=<your-secret-id>, src=<path-to-secret> .` See [Docker documentation](https://docs.docker.com/build/building/secrets/#secret-mounts). |
```
Comment (lzdanski) [inline astro/cli/astro-dev-upgrade-test.md:36]: For this page, `astro dev restart`, `astro dev start`, `astro dev pytest`, and `astro dev parse`, we have that all the options work for both software and astro. If this command doesn't work with software, then we need to shift these pages to the tab structure :/ If they do work for both software + astro, then it also needs to be added to the software tab in `astro deploy`

**Review by lzdanski — COMMENTED** [review]
Important question about this command's behavior for software vs astro and the CLI ref.

File: astro/cli/astro-dev-upgrade-test.md (hunk 4 above)
Comment (sunkickr) [reply astro/cli/astro-dev-upgrade-test.md:36]: This works for both Software and astro

File: astro/cli/private-python-packages.md (hunk 3 above)
Comment (sunkickr) [reply astro/cli/private-python-packages.md:166]: From what I know about the docker feature what you did is correct. But Iv only read the docs. We don't have a private pypi repo to test with. Those using a private pypi repo should be able to just switch out for their information

**Review by sunkickr — APPROVED** [review]

File: astro/cli/release-notes.md [hunk 5]
```diff
@@ -15,6 +15,23 @@ This document provides a summary of all changes made to the [Astro CLI](cli/over
 
 - **Stable versions**: 1.21.0, 1.20.1, and 1.19.4. See [Astro CLI release and lifecycle policy](cli/release-lifecycle-policy.md).
 
+## Astro CLI 1.22.0
+
+Release date: January 24, 2023
+
+### New flag to mount secrets to Astro project image
+
+Use the new `--build-secrets` flag with the following commands to mount a secret value to your Astro project image:
```
Comment (jwitz) [inline astro/cli/release-notes.md:24]:
```suggestion
Use the new `--build-secrets` flag with the following commands to mount a secret value to an Astro project image:
```

**Pull request merged**